sky/data/storage.py

```diff
@@ -1865,10 +1899,574 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+                assert self.name == data_utils.split_cos_path(self.source)[0], (
+                    'COS Bucket is specified as path, the name should be '
+                    'the same as COS bucket.')
+                assert data_utils.verify_ibm_cos_bucket(self.name), (
+                    f'Source specified as {self.source}, a COS bucket. ',
+                    'COS Bucket should exist.')
+        # Validate name
+        self.name = self.validate_container_name(self.name)
+
+        # Check if the storage is enabled
+        if not _is_storage_cloud_enabled(str(clouds.Azure())):
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.ResourcesUnavailableError(
+                    'Storage \'store: azure\' specified, but ' \
+                    'Azure access is disabled. To fix, enable '\
+                    'Azure by running `sky check`. More info: '\
+                    'https://skypilot.readthedocs.io/en/latest/getting-started/installation.html.' # pylint: disable=line-too-long
+                    )
+
+    @classmethod
+    def validate_container_name(cls, name) -> str:
+        """Validates the name of the AZ Container.
+
+        Source for rules: https://learn.microsoft.com/en-us/rest/api/storageservices/Naming-and-Referencing-Containers--Blobs--and-Metadata#container-names # pylint: disable=line-too-long
+
+        Args:
+            name: str; name of the container
+
+        Returns:
+            name: str; name of the container
+
+        Raises:
+            StorageNameError: if the given container name does not follow the
+                naming convention
+        """
+
+        def _raise_no_traceback_name_error(err_str):
+            with ux_utils.print_exception_no_traceback():
+                raise exceptions.StorageNameError(err_str)
+
+        if name is not None and isinstance(name, str):
+            if not 3 <= len(name) <= 63:
+                _raise_no_traceback_name_error(
+                    f'Invalid store name: name {name} must be between 3 (min) '
+                    'and 63 (max) characters long.')
+
+            # Check for valid characters and start/end with a letter or number
+            pattern = r'^[a-z0-9][-a-z0-9]*[a-z0-9]$'
+            if not re.match(pattern, name):
+                _raise_no_traceback_name_error(
+                    f'Invalid store name: name {name} can consist only of '
+                    'lowercase letters, numbers, and hyphens (-). '
+                    'It must begin and end with a letter or number.')
+
+            # Check for two adjacent hyphens
+            if '--' in name:
+                _raise_no_traceback_name_error(
+                    f'Invalid store name: name {name} must not contain '
+                    'two adjacent hyphens.')
+
+        else:
+            _raise_no_traceback_name_error('Store name must be specified.')
+        return name
+
+    def initialize(self):
```

Comment:
This function is too big to read. Is it possible to reference to other storage's implementation to separate it into multiple functions for better readability?

Reply:
Done at [720fe38](https://github.com/skypilot-org/skypilot/pull/3032/commits/720fe38ccd3699ec7cee2e6e7f0966f775a430fd)

Reply:
After putting more thoughts into this, I think it is better to have a single deep `get` function that returns `storage_account_name`, `resource_group_name` values for better readability, so I refactored to do so at [3b771dc](https://github.com/skypilot-org/skypilot/pull/3032/commits/3b771dc1be0d328698bfd89db56ee5f6b7faf04d)